src/index.js

```diff
@@ -53,15 +53,10 @@ class ServerlessPlugin {
       }
 
       this.logMessage(`Validating '${name}' federated graphql schema...`);
-      await apollo.run([
-        'service:push',
-        `--graph=${name}`,
-        `--variant=${variant}`,
-        `--serviceName=${service}`,
-        `--serviceURL=${url}`,
-        `--localSchemaFile=${schema}`,
-        `--key=${apolloKey}`
-      ]);
+      process.env.APOLLO_KEY = apolloKey;
+      exec.execSync(`npx rover subgraph publish ${name}@${variant} --schema ${schema} --name ${service} --routing-url ${url}`, {
```

Comment:
hmm, is this vulnerable to injections? what if `name` was `; echo $AWS_SECRET_KEY` ?

Reply:
Could be however, in this case the user is always going to be the service owner and none of the arguments come from client input. You could just as well add an npm script to the build to echo AWS_SECRET_KEY too.